pkg/common/cluster_domain.go

```diff
@@ -6,11 +6,13 @@ import (
 	"strings"
 )
 
+var ErrCantFindResolvConf = errors.New("failed to parse cluster domain from resolv.conf")
+
 // GetClusterDomain get cluster domain: cluster.local from /etc/resolv.conf
 func GetClusterDomain() (string, error) {
 	resolveConf, err := os.ReadFile("/etc/resolv.conf")
 	if err != nil {
-		return "", err
+		return "", ErrCantFindResolvConf
```

Comment:
Can we better have another error to distinguish between `/etc/resolve.conf` not found or `failed to parse /etc/resolve.conf's content`?

Reply:
I think it's good to make no change rather than `err`.